cmd/iptables-wrapper.go

```diff
@@ -0,0 +1,174 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+)
+
+var (
+	regexChain   = regexp.MustCompile(`(?s):(KUBE-IPTABLES-HINT|KUBE-KUBELET-CANARY)`)
+	regexIPTVers = regexp.MustCompile(`(?s)v1.8.[0123]`)
+)
+
+const (
+	// As we are going directly to distroless, we can assume the binary will be at
+	// `/usr/sbin`
+	iptablesNFTPath           = "/usr/sbin/xtables-nft-multi"
+	iptablesLegacyPath        = "/usr/sbin/xtables-legacy-multi"
+	iptablesNFTMode    string = "nft"
+	iptablesLegacyMode string = "legacy"
+)
+
+// TODO: Move this comment to a better place
+/*
+Assuming that when assembling the container the following binaries will be linked to
+this wrapper:
+* iptables
+* iptables-wrapper
+* iptables-save
+* iptables-restore
+* ip6tables
+* ip6tables-save
+* ip6tables-restore
+
+1) This binary will try to detect which iptables mode was being used by kubelet,
+calling `xtables-<mode>-multi` and checking if the kubelet rules exists in this path
+2) Determined the right mode/binary, it will call `xtables-<mode>-multi` passing
+the command and the arguments to it.
+*/
+func main() {
+	if err := checkIptablesNFTVersion(); err != nil {
+		fmt.Fprintf(os.Stderr, err.Error())
+		os.Exit(1)
+	}
+
+	iptablesMode, err := getIptablesMode()
```

Comment:
```suggestion
	iptablesMode, err := getIPTablesMode()
```

capitalized `iptables` should be `IPTables`